server/chat/backend/agent/tools/memory_tool.py

```
@@ -0,0 +1,557 @@
+# ---------------------------------------------------------------------------
+
+
+class GrepMemoriesArgs(BaseModel):
+    query: str = Field(description="Search pattern — supports regex (e.g. 'redis.*timeout', '(5xx|500)'). Plain strings work too.")
+    category: str = Field(
+        default="",
+        description="Optionally limit search to a specific category",
+    )
+
+
+def grep_memories(
```

Ruff (0.15.18)

[warning] Too many branches (16 > 12)

(PLR0912)